runtime/vm/createramclass.cpp

```diff
@@ -2361,15 +2355,12 @@ internalCreateRAMClassDone(J9VMThread *vmThread, J9ClassLoader *classLoader, J9C
 			classFlags |= J9ClassIsValueType;
 #if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
 			if (J9ROMCLASS_IS_PRIMITIVE_VALUE_TYPE(romClass)
-				|| (J9_ARE_ALL_BITS_SET(romClass->optionalFlags, J9_ROMCLASS_OPTINFO_IMPLICITCREATION_ATTRIBUTE)
-				&& J9_ARE_ALL_BITS_SET(getImplicitCreationFlags(romClass), J9AccImplicitCreateHasDefaultValue))
+				|| J9_ARE_ALL_BITS_SET(classFlags, J9ClassAllowsInitialDefaultValue)
 			) {
 				UDATA instanceSize = state->ramClass->totalInstanceSize;
-				/* This ram class flag is not correct for the nullrestricted value class case
-				 * since the meaning of primitive and nullrestricted don't fully overlap.
-				 * Using it for now to enable flattening of nullrestricted fields.
-				 */
-				classFlags |= J9ClassIsPrimitiveValueType;
+				if (J9ROMCLASS_IS_PRIMITIVE_VALUE_TYPE(romClass)) {
+					classFlags |= J9ClassIsPrimitiveValueType;
```

Comment:
I guess `J9ClassIsPrimitiveValueType` will never be set in lw5 as `J9ROMCLASS_IS_PRIMITIVE_VALUE_TYPE()` is always false there. This bit becomes free in lw5. Can we rename and reuse this bit ?

Reply:
Yes it can be reused but as long as we are still supporting q types I think it would be best to keep them separate to avoid problems [such as this](https://github.com/eclipse-openj9/openj9/pull/18597#issuecomment-1879098039) where this flag caused the ram class to have a q signature. What do you think?

Reply:
After applying this change, are there more test failures in lw5 or less failures ? 

I am a little worried about the places in JIT where `J9_IS_J9CLASS_PRIMITIVE_VALUETYPE` is currently used. As the this PR is updating the places in the VM only, which creates inconsistency between VM and JIT.  Eventually only one of J9ClassAllowsInitialDefaultValue and J9ClassIsPrimitiveValueType will stay. 

> to avoid problems https://github.com/eclipse-openj9/openj9/pull/18597#issuecomment-1879098039 where this flag caused the ram class to have a q signature.

We may ignore the failures regarding Q types, which will be removed eventually.

Reply:
> After applying this change, are there more test failures in lw5 or less failures ?
Sorry do you mean the changes in this pull request or reusing the bit for J9ClassIsPrimitiveValueType?

I can try reusing it. I also worry it will mean we miss adding a check for implicitcreation in a place where J9_IS_J9CLASS_PRIMITIVE_VALUETYPE is expected which could be confusing later.

Reply:
> Sorry do you mean the changes in this pull request or reusing the bit for J9ClassIsPrimitiveValueType?

The changes in this pull request.

Reply:
I would say less, there are 4 tests from ValueTypeTests and 1 from ValueTypeArrayTests (testValueTypeArrayAssignments) that are fixed by this change.

There is one new failure in ValhallaAttributeTests and one new failure in ValueTypeArrayTests for the flattened array option. More details on these from the a previous comment.

Reply:
Hey @hangshao0 I ran the full set of tests for both scenarios, one where J9ClassIsPrimitiveValueType uses a different value to J9ClassAllowsInitialDefaultValue and one where they are both the same. The case where they are the same has slightly fewer failures so I've updated the change for them to use the same flag.